catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ValidateCommand.java

```diff
@@ -47,178 +55,104 @@
-        // so we want to make the last element of the list pronounced so that we can more
-        // easily see the groupings of the hierarchy. This is why we use an iterator
-        // and if-else rather than a more simple for-loop.
-        Iterator<MetacardValidator> iterator = validators.iterator();
-        while (iterator.hasNext()) {
-            MetacardValidator validator = iterator.next();
-            if (iterator.hasNext()) {
-                printValidator(validator, ELEMENT);
-                printErrorsAndWarnings(validator, metacard, INDENT);
-            } else {
-                printValidator(validator, LAST_ELEMENT);
-                printErrorsAndWarnings(validator, metacard, LAST_INDENT);
-            }
-        }
-    }
+    private List<Metacard> getMetacardsFromCatalog() throws Exception {
+        List<Metacard> results = new ArrayList<>();
 
-    private void printValidator(MetacardValidator validator, String prefix) {
-        String name = validator.getClass()
-                .getName();
-        if (validator instanceof Describable && ((Describable) validator).getId() != null) {
-            name = ((Describable) validator).getId();
-        }
-        prettyPrint(prefix, name);
-    }
+        Filter filter = builder.attribute(Metacard.ANY_TEXT)
+                .text(query);
 
-    private void printErrorsAndWarnings(MetacardValidator validator, Metacard metacard,
-            String prefix) {
-        try {
-            validator.validate(metacard);
-            prettyPrint(prefix + LAST_ELEMENT, "No errors or warnings");
-        } catch (ValidationException e) {
-
-            // The seemingly unnecessary complexity here is because we want the end result
-            // on the console to look good. We know that errors will come before warnings,
-            // so the whole error block uses the "indent" prefix. Further, we print
-            // errors and warnings with different
-            prettyPrint(prefix + ELEMENT, ERROR_TITLE);
-            if (e.getErrors() == null) {
-                prettyPrint(prefix + INDENT + LAST_ELEMENT, "None");
-            } else {
-                printList(prefix + INDENT, e.getErrors(), ERROR_COLOR);
-            }
+        QueryImpl query = new QueryImpl(filter);
+        ThreadContext.bind(Security.getInstance()
+                .getSystemSubject());
 
-            // We know that warnings come last, so we format each line knowing that it's the last
-            // in its hierarchy.
-            prettyPrint(prefix + LAST_ELEMENT, WARNING_TITLE);
-            if (e.getWarnings() == null) {
-                prettyPrint(prefix + LAST_INDENT + LAST_ELEMENT, "None");
-            } else {
-                printList(prefix + LAST_INDENT, e.getWarnings(), WARNING_COLOR);
+        SourceResponse response = catalog.query(new QueryRequestImpl(query));
+        List<Result> resultList = response.getResults();
+        if (resultList != null && resultList.size() > 0) {
```

Comment:
I wish our API didn't return null for collections. Then you could eliminate this conditional entirely, since a for-each over an empty collection is a no-op.